flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/table/StateConnectorOptions.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.table;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.configuration.description.Description;
+
+import static org.apache.flink.configuration.description.TextElement.code;
+
+/** Options for the state connector. */
+@PublicEvolving
+public class StateConnectorOptions {
+
+    public static final String FIELDS = "fields";
+    public static final String STATE_NAME = "state-name";
+    public static final String STATE_TYPE = "state-type";
+    public static final String MAP_KEY_FORMAT = "map-key-format";
+    public static final String VALUE_FORMAT = "value-format";
+
+    /** Value state types. */
+    public enum StateType {
+        VALUE,
+        LIST,
+        MAP
+    }
+
+    // --------------------------------------------------------------------------------------------
+    // Common options
+    // --------------------------------------------------------------------------------------------
+
+    public static final ConfigOption<String> STATE_BACKEND_TYPE =
+            ConfigOptions.key("state.backend.type")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Defines the state backend which must be used for state reading.");
```

Comment:
I think the valid values should be included in the description, as per the Flip - I assume with ForsDB as well.

Reply:
This is similar like [state.backend.type](https://github.com/apache/flink/blob/9ba5afc26057672fd0491bb3f3e50a55ce1c32b1/flink-core/src/main/java/org/apache/flink/configuration/StateBackendOptions.java#L47) so I would keep it that way.

Reply:
Adding more description makes sense though.